File: Python/Data structure/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        lookup = {}
        for i, val in enumerate(nums):
        	if (target - val) in lookup:
        		return [ lookup[target-val], i]
        	else:
        		lookup[val] = i


print Solution().twoSum( [2,7,11,15], 9)

File: src/main/java/com/samjin/Company/Amazon/TwoSum.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.samjin.Company.Amazon;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

/**
 *
 * @author Sam
 */
public class TwoSum {
    
    public static int[] twoSum(int[] nums, int target) {
        
        Map<Integer,Integer> map = new HashMap<>();
        
        for( int i = 0; i < nums.length; i++ ){
            if( map.containsKey( nums[i]) ){
                return new int[]{ map.get(nums[i]), i };
            }else{
                map.put( target-nums[i], i );
            }
        }
        return new int[]{0,0};
    }
    
    public static void main(String[] args) {
        int[] test = { 2, 7, 11, 15 };
        System.out.println( Arrays.toString( twoSum(test,9) ));
    }
}
